File: convalida-validators/src/test/java/convalida/validators/BaseTest.java
package convalida.validators;

import android.text.Editable;
import android.widget.EditText;
import org.junit.Before;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

/**
 * @author Wellington Costa on 16/11/2017.
 */
public class BaseTest {

    protected final String EMPTY_VALUE = "";

    protected EditText mockEditText;

    protected final String errorMessage = "Field invalid";

    @Before public void setUp() {
        mockEditText = mock(EditText.class);
        when(mockEditText.getText()).thenReturn(mock(Editable.class));
    }

}

File: convalida-validators/src/test/java/convalida/validators/BaseTest.kt
package convalida.validators

import android.text.Editable
import android.widget.EditText
import org.junit.Before
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock

/**
 * @author Wellington Costa on 16/11/2017.
 */
open class BaseTest {

    internal lateinit var mockEditText: EditText
    internal val errorMessage = "Field invalid"

    @Before
    fun setUp() {
        mockEditText = mock(EditText::class.java)
        `when`(mockEditText.text).thenReturn(mock(Editable::class.java))
    }

}
